From dd6631a0b99bc3dcbf6129b535ab898b4d54d929 Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Fri, 10 Jul 2026 06:15:28 +0200
Subject: [PATCH] Revert "drm/amdgpu: Use amdgpu by default for CIK APUs too"

This reverts commit d1f188b182c92b71d1bdd03436d7c6b08fbc86be, breaks on the HP t520.
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c | 7 +++++--
 drivers/gpu/drm/radeon/radeon_drv.c     | 3 ++-
 2 files changed, 7 insertions(+), 3 deletions(-)

--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -641,7 +641,9 @@ module_param_named(si_support, amdgpu_si
  * CIK (Sea Islands) are second generation GCN GPUs, supported by both
  * drivers: radeon (old) and amdgpu (new). This parameter controls whether
  * amdgpu should support CIK.
- * By default, CIK dedicated GPUs and APUs are supported by amdgpu.
+ * By default:
+ * - CIK dedicated GPUs are supported by amdgpu.
+ * - CIK APUs are supported by radeon (except when radeon is not built).
  * Only relevant when CONFIG_DRM_AMDGPU_CIK is enabled to build CIK support in amdgpu.
  * See also radeon.cik_support which should be disabled when amdgpu.cik_support is
  * enabled, and vice versa.
@@ -2321,6 +2323,8 @@ static bool amdgpu_support_enabled(struc
 
 	case CHIP_BONAIRE:
 	case CHIP_HAWAII:
+		support_by_default = true;
+		fallthrough;
 	case CHIP_KAVERI:
 	case CHIP_KABINI:
 	case CHIP_MULLINS:
@@ -2328,7 +2332,6 @@ static bool amdgpu_support_enabled(struc
 		param = "cik_support";
 		module_param = amdgpu_cik_support;
 		amdgpu_support_built = IS_ENABLED(CONFIG_DRM_AMDGPU_CIK);
-		support_by_default = true;
 		break;
 
 	default:
--- a/drivers/gpu/drm/radeon/radeon_drv.c
+++ b/drivers/gpu/drm/radeon/radeon_drv.c
@@ -278,13 +278,14 @@ static bool radeon_support_enabled(struc
 
 	case CHIP_BONAIRE:
 	case CHIP_HAWAII:
+		support_by_default = false;
+		fallthrough;
 	case CHIP_KAVERI:
 	case CHIP_KABINI:
 	case CHIP_MULLINS:
 		gen = "CIK";
 		module_param = radeon_cik_support;
 		amdgpu_support_built &= IS_ENABLED(CONFIG_DRM_AMDGPU_CIK);
-		support_by_default = false;
 		break;
 
 	default:
